rfcs/20230323-add-split-graph-op/README.md

```diff
@@ -0,0 +1,161 @@
+| attribute | `num_splits`  | number of outputs into which the input tensor data will be split along axis dimension. | optional             | s64           |
+| attribute | `size_splits` | list of sizes for each output alone axis dims.                                         | optional             | s64           |
+| output    | `dst_i`       | output data                                                                            | Required             | f32,f16,bf16* |
+
+@note: 
+- input and outputs must have the same data type.
+-The i-th output has the same shape as src input tensor except for dimension
+along `axis` which is src.dims[`axis`]/`num_splits` or `size_splits`[i].
+- Either `num_splits` or `size_splits` should be provided. When `num_splits` is 
+used,`size_splits` will be ignored.
+- If `num_splits` is specified, the dimension of input tensor data shape along 
+`axis` must be evenly divisible by `num_splits`.
+- If `size_splits` is specified, the sum of the values in `size_splits` must be 
+equal to the src.dims[`axis`].
+
+Pros.
+1. The definition is simple and we only focus on the static shape support.
+
+Cons.
+1. We may need to add a dynamic split in the future when graph API support dynamic shape.
+2. Not suitable for Pytorch integration to map `split_size_or_sections ` of `Split`
+in Pytorch to `num_splits`.
+
+### Option2: 
+
+`Split` operation supports four attributes: `axis`, `num_splits` , `size_split`
+and `size_splits`. `axis`, `num_splits` and `size_split` should be an interger 
+value. `size_splits` is a 1-D Tensor (or list). The input data is split into 
+`num_splits` or len(`size_splits`) elements. The shape of the i-th element has 
+the same size as the value except along dimension `axis` where the size is 
+determined by one of `num_splits`, `size_split`, `size_splits`. Note that Split
+operation is not guaranteed to return a view or a copy of original input tensor.
+
+| Split     | Argument Name | Description                                                                            | Required or Optional | Value Type    |
+|-----------|---------------|----------------------------------------------------------------------------------------|----------------------|---------------|
+| input     | `src`         | input data                                                                             | Required             | f32,f16,bf16* |
+| attribute | `axis`        | Specifies which dimension to split along.                                              | Required             | s64           |
+| attribute | `num_splits`  | number of outputs into which the input tensor data will be split along axis dimension. | optional             | s64           |
+| attribute | `size_split`  | size for single output tensor along axis dimension.                                    | optional             | s64           |
```

Comment:
during graph build stage, what if `size_split` is specified while input shape is unknown? How to determine the output number?

Reply:
That case you mentioned is a dynamic case that is not supported and has not been encountered yet. Once the dynamic shape feature lands in the future release, we can take it into consideration. cc@wuxun-zhang. 
The option2 here is for the convenience of integration on the IPEX side which can simply map the `size_split_or_sections` into `size_split` during the build stage.

Reply:
I think it's not a dynamic shape case. FWK won't provide concrete shapes during the graph build stage even for models with static shapes.

Reply:
> during graph build stage, what if `size_split` is specified while input shape is unknown? How to determine the output number?

Why need to determine the output number at graph build stage?

Reply:
> Why need to determine the output number at graph build stage?

If you don't know the output number, how to build the graph after this op? A simple example is split+concat, how to collect all the outputs from split and feed them into contact op?

Reply:
I see, it's related to how bridge pass framework's graph to oneDNN Graph. We may need framework team to help answer, cc @chunyuan-w @sanchitintel @retonym.

Reply:
Do I understand correctly that this is a case where the graph structure depends on the input shapes? In the current integration, the graph will only be built once. I feel like it does not work anymore in such cases. Let me check with @jgong5 & @sanchitintel on it.

Reply:
@chunyuan-w, I agree that this case would encounter a runtime error with our multiple compiled partitions per partition approach. I think we can't solve this problem without re-creating the graph.